private-media-framework/draft-ietf-perc-private-media-framework.md

```diff
@@ -735,71 +757,77 @@ end-to-end encrypted in PERC.
 SRTP master key as a result of the EKT key changing.
 
 A locally-generated SRTP master key is used along with the master salt
-transmitted to the endpoint from the key distributor via the EKTKey
+transmitted to the Endpoint from the Key Distributor via the EKTKey
 message to encrypt media end-to-end.
 
-Since the media distributor is not involved in E2E functions, it does not
-create this key nor have access to any endpoint's E2E key.  Note, too,
-that even the key distributor is unaware of the locally-generated E2E keys
-used by each endpoint.
+Since the Media Distributor is not involved in E2E functions, it does not
+create this key nor have access to any Endpoint's E2E key.  Note, too,
+that even the Key Distributor is unaware of the locally-generated E2E keys
+used by each Endpoint.
 
-The endpoint transmits its E2E key to other endpoints in the conference
+The Endpoint transmits its E2E key to other Endpoints in the conference
 by periodically including it in SRTP packets in a Full EKT Tag.  When
 placed in the Full EKT Tag, it is encrypted using the EKT Key provided
-by the key distributor.  The master salt is not transmitted, though,
-since all endpoints receive the same master salt via the EKTKey
+by the Key Distributor.  The master salt is not transmitted, though,
+since all Endpoints receive the same master salt via the EKTKey
 message from the Key Distributor.  The recommended frequency with which an
-endpoint transmits its SRTP master key is specified in
+Endpoint transmits its SRTP master key is specified in
 [@!I-D.ietf-perc-srtp-ekt-diet].
 
 ## Summary of Key Types and Entity Possession
 
-All endpoints have knowledge of the KEK.
+All Endpoints have knowledge of the KEK.
 
-Every HBH key is distinct for a given endpoint, thus Endpoint A and
-Endpoint B do not have knowledge of the other's HBH key.
+Every HBH key is distinct for a given Endpoint, thus Endpoint A and
+Endpoint B do not have knowledge of the other's HBH key.  Since HBH keys
+are derived from a DTLS-SRTP association, there is at most one HBH key
+per Endpoint,  (The only exception is where the DTLS-SRTP association might
+be rekeyed per Section 5.2 of [@RFC5764] and a new key is created to
+replace the former key.)
 
-Each endpoint generates its own E2E Key (SRTP master key), thus
-distinct per endpoint.  This key is transmitted (encrypted) via
-the Full EKT Tag to other endpoints.  Endpoints that receive media from
-a given transmitting endpoint gains knowledge of the
-transmitter's E2E key via the Full EKT Tag.
+Each Endpoint generates its own E2E key (SRTP master key), thus
+there is a distinct E2E key per endpoint.  This key is transmitted (encrypted) via
+the Full EKT Tag to other Endpoints.  Endpoints that receive media from
+a given transmitting Endpoint gain knowledge of the
+transmitter's E2E key via the Full EKT Tag.  An Endpoint **MAY** create
+a unique E2E key per media flow, where a media flow is uniquely identified
+by its SSRC value.
```

Comment:
just checking, why was the last sentence added ?

Reply:
Because I felt like this was causing some confusion.  The endpoint may create a key per flow, if it wants.  This is compatible with existing language like this from the draft:

```
An Endpoint will receive one or more E2E keys from
every other Endpoint in the conference that correspond to the media flows
transmitted by those other Endpoints
```

Maybe I should move that to section 4.5 (Key Exchange)?  I do want to remove the word "unique" as that is not a requirement.  Plus, there may be one per SPI.

Should I remove this?  Move it to 4.5?  Or replicate the idea in 4.5?

Reply:
I feel per ssrc unique key is RFC3711 and not PERC specific.. isn't it ?

Reply:
Because we speak so often of using DTLS-SRTP to get "a" key, I think it's important to note the endpoint may use multiple keys it creates, even if that might seem redundant with SRTP.